tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js

```diff
@@ -77,6 +104,102 @@ module.exports = {
       }
       return this.api.page.FilesPageElement.filesList()
     },
+    /**
+     *
+     * @param {string} collaborator
+     * @param {string} requiredPermissions
+     */
+    changeCustomPermissionsTo: async function (collaborator, requiredPermissions) {
+      const requiredPermissionArray = this.getArrayFromPermissionString(requiredPermissions)
+      const informationSelector = util.format(this.elements.collaboratorInformationByCollaboratorName.selector, collaborator)
+      const selectRoleButton = informationSelector + this.elements.selectRoleButtonInCollaboratorInformation.selector
+
+      let permission = ''
+      this
+        .useXpath()
+        .waitForElementVisible(informationSelector)
+        .click(informationSelector)
+        .waitForElementVisible(selectRoleButton)
+
+      for (let i = 0; i < collaboratorPermissionArray.length; i++) {
+        permission = collaboratorPermissionArray[i]
+        const permissionXpath = this.getPermissionSwitchXpath(permission)
+        // Check if the xpath of permission is visible
+        await this.api.waitForElementVisible({ selector: permissionXpath, timeout: 100, abortOnFailure: false }, function (result) {
+          if (result.value === true) {
+            this
+              .getAttribute(permissionXpath, 'data-state', (state) => {
+                if ((state.value === 'on' && !requiredPermissionArray.includes(permission)) ||
+                  (state.value === 'off' && requiredPermissionArray.includes(permission))) {
+                  // need to click
+                  this.useXpath()
+                    .waitForElementVisible(permissionXpath)
+                    .click(permissionXpath)
+                }
+              })
+          }
+          // check if the requiredPermission is not visible
+          if (result.value === false && requiredPermissionArray.includes(permission)) {
+            throw new Error(`permission ${permission} is not visible `)
+          }
+        })
+      }
+
+      return this.waitForElementVisible('@saveShareButton')
+        .click('@saveShareButton')
+        .waitForOutstandingAjaxCalls()
+        .waitForElementNotPresent('@saveShareButton')
+    },
+    /**
+     *
```

Comment:
please add a comment like:

```suggestion
     * asserts that the permission is set to "off" or not displayed at all
```

because this function would only throw an exception if its visible and on